engine/access/ingestion/engine_test.go

```diff
@@ -134,28 +143,26 @@ func (suite *Suite) TestOnFinalizedBlock() {
 	for _, guarantee := range block.Payload.Guarantees {
 		needed[guarantee.ID()] = struct{}{}
 	}
+
+	wg := sync.WaitGroup{}
+	wg.Add(4)
+
 	suite.request.On("EntityByID", mock.Anything, mock.Anything).Run(
 		func(args mock.Arguments) {
 			collID := args.Get(0).(flow.Identifier)
 			_, pending := needed[collID]
 			suite.Assert().True(pending, "collection should be pending (%x)", collID)
 			delete(needed, collID)
+			wg.Done()
 		},
 	)
 
 	// process the block through the finalized callback
 	suite.eng.OnFinalizedBlock(&hotstuffBlock)
-
-	// wait for engine shutdown
-	done := suite.eng.unit.Done()
```

Comment:
I think you'll still need to shutdown the engine at the end of the test, otherwise it will leak go routines. Similar to what you did in `TestComponentShutdown`